crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -1088,42 +1068,37 @@ impl<InstalledPackages: InstalledPackagesProvider> ResolverState<InstalledPackag
                     }
                 };
 
-                let mut dependencies = PubGrubDependencies::from_requirements(
+                let requirements = self.flatten_requirements(
                     &metadata.requires_dist,
                     &metadata.dev_dependencies,
-                    &self.constraints,
-                    &self.overrides,
-                    Some(name),
                     extra.as_ref(),
                     dev.as_ref(),
+                    Some(name),
+                    markers,
+                );
+
+                let mut dependencies = PubGrubDependencies::from_requirements(
+                    &requirements,
+                    Some(name),
                     &self.urls,
                     &self.locals,
                     &self.git,
-                    self.markers.as_ref(),
-                    self.requires_python.as_ref(),
-                    markers,
-                )?;
-
-                for (dep_package, dep_version) in dependencies.iter() {
-                    debug!("Adding transitive dependency for {package}=={version}: {dep_package}{dep_version}");
-
-                    // Update the package priorities.
-                    priorities.insert(dep_package, dep_version);
-
-                    // Emit a request to fetch the metadata for this package.
-                    self.visit_package(dep_package, request_sink)?;
-                }
+                );
 
                 // If a package has metadata for an enabled dependency group,
                 // add a dependency from it to the same package with the group
                 // enabled.
-                if extra.is_none() && dev.is_none() {
-                    for dev in &self.dev {
-                        if metadata.dev_dependencies.contains_key(dev) {
+                // TODO(konsti): Why don't we eagerly pull in dev deps, like we do with extras?
```

Comment:
What does this mean? Can I help explain it?

Reply:
Yes please, my question is: For the extras, we add the actual requirements from the extra to the dependencies; for dev-dependencies, why do we only add the proxy package?